src/frequenz/sdk/timeseries/ev_charger_pool/_ev_charger_pool.py

```diff
@@ -5,147 +5,51 @@
-            A tuple with the component ID of an ev charger that just had a state
-                change, and its new state.
-        """
-        if self._changed_component is None:
-            return None
-        return (
-            self._changed_component,
-            self._states.setdefault(
-                self._changed_component, EVChargerState.UNSPECIFIED
-            ),
-        )
-
-
-class _StateTracker:
-    """A class for keeping track of the states of all ev chargers in a pool."""
-
-    def __init__(self, comp_states: dict[int, EVChargerState]) -> None:
-        """Create a `_StateTracker` instance.
-
-        Args:
-            comp_states: initial states of all ev chargers in the pool.
-        """
-        self._states = comp_states
-
-    def get(self) -> EVChargerPoolStates:
-        """Get a representation of the current states of all ev chargers.
-
-        Returns:
-            An `EVChargerPoolStates` instance.
-        """
-        return EVChargerPoolStates(self._states)
-
-    def update(
-        self,
-        data: EVChargerData,
-    ) -> Optional[EVChargerPoolStates]:
-        """Update the state of an ev charger, from a new data point.
-
-        Args:
-            data: component data from the microgrid, for an ev charger in the pool.
-
-        Returns:
-            A new `EVChargerPoolStates` instance representing all the ev chargers in
-                the pool, in case there has been a state change for any of the ev
-                chargers, or `None` otherwise.
-        """
-        evc_id = data.component_id
-        new_state = EVChargerState.from_ev_charger_data(data)
-        if evc_id not in self._states or self._states[evc_id] != new_state:
-            self._states[evc_id] = new_state
-            return EVChargerPoolStates(self._states, evc_id)
-        return None
+class EVChargerPoolError(Exception):
+    """An error that occurred in any of the EVChargerPool methods."""
 
 
 class EVChargerPool:
     """Interactions with EV Chargers."""
 
     def __init__(
         self,
-        component_ids: Optional[set[int]] = None,
+        channel_registry: ChannelRegistry,
+        resampler_subscription_sender: Sender[ComponentMetricRequest],
+        component_ids: set[int] | None = None,
     ) -> None:
         """Create an `EVChargerPool` instance.
 
         Args:
+            channel_registry: A channel registry instance shared with the resampling
+                actor.
+            resampler_subscription_sender: A sender for sending metric requests to the
+                resampling actor.
             component_ids: An optional list of component_ids belonging to this pool.  If
                 not specified, IDs of all ev chargers in the microgrid will be fetched
                 from the component graph.
         """
-        self._component_ids = set()
+        self._channel_registry = channel_registry
+        self._resampler_subscription_sender = resampler_subscription_sender
```

Comment:
This attribute name is too long

Reply:
I like to keep the attribute name the same as the parameter,  but related to: https://github.com/frequenz-floss/frequenz-sdk-python/pull/201#discussion_r1116812353

so it can be changed separately, later.

Reply:
This will be kept as it is